engine/Assets/Scripts/UI/Dynamic/Modals/Configuring/ConfigMotorModal.cs

```diff
@@ -0,0 +1,250 @@
+        }
+
+        // original target velocities for each motor and entry in scrollview for other motors
+        for (i = 0; i < _motors.Length; i++) {
+            var driver = _motors[i].driver;
+            switch (driver) {
+                case (RotationalDriver):
+                    _motors[i].origVel = (driver as RotationalDriver).Motor.targetVelocity;
+                    break;
+                case (WheelDriver):
+                    _motors[i].origVel = (driver as WheelDriver).Motor.targetVelocity;
+                    break;
+                case (LinearDriver):
+                    _motors[i].origVel = (driver as LinearDriver).MaxSpeed;
+                    break;
+            }
+
+            if (_motors[i].motorType == MotorType.Other) {
+                int j = i;
+                var u = "RPM";
+                if (_motors[i].driver is LinearDriver) u = "M/S";
+                CreateEntry(GetName(_motors[i].driver), _motors[j].origVel, x => _motors[j].setTargetVelocity(x), u);
+            }
+        }
+    }
+
+    public override void Update() {}
+
+    public override void Delete() {}
+
+    private void CreateEntry(string name, float currVel, Action<float> onClick, string units = "RPM") {
+        (Content nameContent, Content velContent) =
+            _scrollView.Content.CreateSubContent(new Vector2(_scrollViewWidth, 40f))
+                .SetTopStretch<Content>(0, 0, 0)
+                .ApplyTemplate<Content>(VerticalLayout)
+                .SplitLeftRight(NAME_WIDTH, PADDING);
+        nameContent.CreateLabel().SetText(name).SetTopStretch(0, PADDING, PADDING + _scrollView.HeightOfChildren);
+        float max = 150;
+        if (currVel < 5) max = 50f;
+        velContent.CreateSlider(units, minValue: 0f, maxValue: max, currentValue: currVel)
+            .SetTopStretch<Slider>(PADDING, PADDING, _scrollView.HeightOfChildren)
+            .AddOnValueChangedEvent((s, v) => { onClick(v); });
+    }
+
+    private void ChangeDriveVelocity(float vel) {
+        foreach (ConfigMotor motor in _motors) {
+            if (motor.motorType == MotorType.Drive)
+                motor.setTargetVelocity(vel);
+        }
+    }
+
+    private void ChangeTurnVelocity(float vel) {
+        foreach (ConfigMotor motor in _motors) {
+            if (motor.motorType == MotorType.Turn)
+                motor.setTargetVelocity(vel);
+        }
+    }
+
+    private class ConfigMotor {
+        public Driver driver;
+        public float origVel;
+        private float _force;
+        public bool velChanged = false;
+        public MotorType motorType;
+
+        public ConfigMotor(MotorType t) {
+            motorType = t;     
+        }
+
+        public void setTargetVelocity(float v) {
+            switch (driver) {
+                case (RotationalDriver):
+                    _force = (driver as WheelDriver).Motor.force;
```

Comment:
Causes an error when changing velocity of arms

Reply:
Fixed. Thanks for catching that. I never got the error, so let me know if it's still there.